### PR TITLE
Feature/808 rounds block be

### PR DESCRIPTION
**Pull request by todd-dembrey**
Rebased onto #857 

Includes some refactoring to make accessing rounds and labs easier for others

### CODE REVIEW

File: opentech/apply/funds/models/applications.py [hunk 1]
```diff
@@ -314,3 +333,101 @@ def get_submit_meta_data(self, **kwargs):
...
+        return qs.filter(progress=100)
+
+    def new(self):
+        return qs.filter(start_date__gt=date.today())
+
+    def open(self):
+        return self.filter(Q(end_date__gte=date.today(), start_date__lte=date.today()) | Q(end_date__isnull=True))
+
+    def closed(self):
+        return self.filter(end_date__lt=date.today())
```
[the remark this answers is not in the document]
Comment (emullaney) [reply opentech/apply/funds/models/applications.py]: I think that they should all be `self.filter`

File: opentech/apply/funds/templates/funds/includes/round-block-listing.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<ul class="round-block">
+    {% for round in rounds %}
+        {% if forloop.counter0 < 5 %}
+            <li class="round-block__item">
+                <h4 class="round-block__title">{{ round }}</h4>
+                <p> {{ round.fund|default_if_none:"-" }} </p>
+                <p class="round-block__date">
+                    {% if round.end_date %}
+                        {{display_text }} {{ round.end_date|date:"Y-m-d" }}
```
Comment (emullaney) [inline opentech/apply/funds/templates/funds/includes/round-block-listing.html]: Nitpick, missing space here before display_text

File: opentech/apply/funds/templates/funds/includes/round-block.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    </section>
+
+    {# Closed rounds/labs tab #}
+    <div class="tabs__content" id="tab-1">
+        {% include "funds/includes/round-block-listing.html" with rounds=closed_rounds display_text="Closed" query=closed_query %}
+    </div>
+
+    {# Open rounds/labs tab #}
+    <div class="tabs__content" id="tab-2">
+        {% include "funds/includes/round-block-listing.html" with rounds=open_rounds display_text="Closed" query=open_query %}
```
Comment (emullaney) [inline opentech/apply/funds/templates/funds/includes/round-block.html]: I think we discussed the text saying "Open until" on this tab

File: opentech/apply/funds/models/applications.py [hunk 4]
```diff
@@ -314,3 +333,101 @@ def get_submit_meta_data(self, **kwargs):
...
+                0,
+            ),
+        ).annotate(
+            progress=Case(
+                When(total_submissions=0, then=None),
+                default=(F('closed_submissions') * 100) / F('total_submissions'),
+                output_fields=FloatField(),
+            )
+
+        )
```
Comment (emullaney) [inline opentech/apply/funds/models/applications.py]: Cool, really nicely done

File: opentech/apply/funds/models/applications.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (todd-dembrey) [reply opentech/apply/funds/models/applications.py]: Good spot, bad refactoring on my part. It got triggered in testing (see sentry below). added tests to cover the new functionality.

https://sentry.io/otf/otf-webapp-test/issues/849136179/

File: opentech/apply/funds/templates/funds/includes/round-block.html (hunk 3 above)
Comment (todd-dembrey) [reply opentech/apply/funds/templates/funds/includes/round-block.html]: copy paste, good catch

**Pull request merged**